### ISSUE TITLE
Custom TableNameResolver have argument Optional.class instead of Entity.class

### ISSUE THREAD

**Issue by LukaszNowacki94** [derjust/spring-data-dynamodb#109]
Hi. I have problem when deleting an entity using deleteById.
I have table name resolver like below that is used to get table name for entity that code is working on 

```java
public class TableNameResolverImpl implements DynamoDBMapperConfig.TableNameResolver {

    private static final String PREFIX_1 = "X";
    private static final String PREFIX_2 = "Y";
    private static final String PREFIX_3 = "Z";
    private String environment;

    public TableNameResolverImpl(final String environment) {
        this.environment = environment;
    }

    @Override
    public String getTableName(final Class<?> aClass, final DynamoDBMapperConfig dynamoDBMapperConfig) {
        return Joiner.on("-").join(PREFIX_1, PREFIX_2, PREFIX_3, aClass.getAnnotation(DynamoDBTable.class).tableName(),
                environment)
                .toUpperCase();
    }
}

```

The problem is that with newer versions of library, findById function return now `Optional<T>` instead of `Entity` or `null`.  And in code when in FindById function instead passing an entity to function `delete(Entity T)` right now is passed `delete(Optional<Entity> T)`. 
Therefore tableNameResolver instead of getting table name 

```
X-Y-Z-EntityClassTableName-environment
```

It throws `NullPointerException`
Basically argument
`final Class<?> aClass` instead having value of Entity.class, it have value of `Optional`.class and it throws NullPointerException because of no annotation DynamoDBTable in class `Optional`.
One of solution that i found for this is small change in `SimpleDynamoDBCrudRepository` to passing Entity without wrapping it in Optional. I've already have fixed version on my local computer with tests, so if you want i can create a PR for quickFix.

**Comment by derjust** [comment]
If you have an fix/code/test for that please go ahead and create a PR - this is _always_ helpful :)

I understand the issue and as you have already done the legwork regarding the research I hope I can get out a fixed `5.0.x` soon

**Comment by derjust** [comment]
Fixed in `5.0.1`
